src/translators/timescale.py

```diff
@@ -180,57 +179,30 @@ def _update_data_table(self, table_name, new_columns, fiware_service):
-            elif cn == FIWARE_SERVICEPATH:
-                values.append(fiware_servicepath or '')
-            else:
-                # Normal attributes
-                try:
-                    attr = original_attrs[cn][0]
-                    attr_t = original_attrs[cn][1]
-                    mapped_type = self._compute_type(e['id'], attr_t, e[attr])
-
-                    if SlfGeometry.is_ngsi_slf_attr(e[attr]):
-                        ast = SlfGeometry.build_from_ngsi_dict(e[attr])
-                        mapped_value = geocoding.slf.wktcodec.encode_as_wkt(
-                            ast, srid=4326)
-                    elif mapped_type == NGSI_TO_SQL[NGSI_GEOJSON]:
-                        mapped_value = geocoding.geojson.wktcodec.encode_as_wkt(
-                            e[attr]['value'], srid=4326)
-                    elif mapped_type == NGSI_TO_SQL[NGSI_STRUCTURED_VALUE]:
-                        mapped_value = pg8000.PGJsonb(e[attr]['value'])
-                    elif mapped_type == NGSI_TO_SQL[NGSI_TEXT] \
-                            and 'value' in e[attr] and e[attr]['value'] is not None:
-                        mapped_value = str(e[attr]['value'])
-                    elif mapped_type == PG_JSON_ARRAY:
-                        mapped_value = pg8000.PGJsonb(e[attr]['value'])
-                    elif 'type' in e[attr] and e[attr]['type'] == 'Property' \
-                            and 'value' in e[attr] \
-                            and isinstance(e[attr]['value'], dict) \
-                            and '@type' in e[attr]['value'] \
-                            and e[attr]['value']['@type'] == 'DateTime':
-                        mapped_value = e[attr]['value']['@value']
-                    elif 'type' in e[attr] and e[attr][
-                            'type'] == 'Relationship':
-                        mapped_value = e[attr].get('value', None) or \
-                                       e[attr].get('object', None)
-                    else:
-                        mapped_value = e[attr]['value']
-
-                    values.append(mapped_value)
-                except KeyError:
-                    # this entity update does not have a value for the column
-                    # so use None which will be inserted as NULL to the db.
-                    values.append(None)
-        return values
+    @staticmethod
+    def _ngsi_geojson_to_db(attr):
+        return geocoding.geojson.wktcodec.encode_as_wkt(attr['value'], srid=4326)
+
+    @staticmethod
+    def _ngsi_slf_to_db(attr):
+        ast = SlfGeometry.build_from_ngsi_dict(attr)
+        return geocoding.slf.wktcodec.encode_as_wkt(ast, srid=4326)
+
+    @staticmethod
+    def _ngsi_structured_to_db(attr):
+        attr_v = attr.get('value', None)
+        if isinstance(attr_v, dict):
+            return pg8000.PGJsonb(attr_v)
+        logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
```

Comment:
Is this going to work in all cases? what if `attr` has no `value` attribute in it? ditto for `type`.